cli/flox-rust-sdk/src/models/environment/generations.rs

```diff
@@ -61,7 +61,11 @@ pub struct ReadOnly {}
 /// Todo: rename to `CheckedOut`, `Filesystem`, ...?
 ///
 /// /// See also: [ReadOnly]
-pub struct ReadWrite {}
+pub struct ReadWrite<'p> {
+    /// A reference to the [ReadOnly] instance that will be released
+    /// when this instance is dropped.
+    _read_only: &'p mut Generations<ReadOnly>,
```

Comment:
nonblocking: why do we need to keep the ReadOnly instance from being dropped?

Reply:
Currently there is no way to get the read only instance back, you convert it to writable, but then it was "gone".
You can either explicitly convert the writable back to read or like here hold an exclusive reference hostage that is released on drop.